src/orange/detail/BIHPartitioner.cc

```diff
@@ -0,0 +1,180 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/detail/BIHPartitioner.cc
+//---------------------------------------------------------------------------//
+#include "BIHPartitioner.hh"
+
+#include "corecel/math/SoftEqual.hh"
+#include "orange/BoundingBoxUtils.hh"
+#include "orange/detail/BIHUtils.hh"
+
+namespace
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sort and uniquify a vector.
+ */
+void sort_and_uniquify(std::vector<celeritas::real_type>& vec)
```

Comment:
I think this should be `fast_real_type` too? It doesn't make sense to uniquify on doubles and then let the truncated floats be identical.

Reply:
Same comment as below. I can change BIHBuilder to simply convert all BoundingBoxes to FastBBox immediately, otherwise this gets messy

Reply:
Thinking about this more: Uniquifying is not strictly necessary anymore now that we are testing only 3 partitions per axis (instead of all possible partitions), so uniquifying on double vs float is not going to make any difference. I think the way I have it with everything being done with doubles until storage is fine as it is. Keep in mind, the only reason we went to floats for boxes was that I proposed that we store them during runtime, and we might NOT even store them during runtime if they don't improve point-in-volume performance.